data/scripts/creaturescripts/player/idle_time.lua

```diff
@@ -0,0 +1,38 @@
+local config = {
+	maxIdleTime = 15 * 60 * 1000, -- 15 minutes
+	kickAfterWarn = 1 * 60 * 1000 -- 1 minute
+}
+
+-- register player event
+local event = CreatureEvent("RegisterIdleTime")
+
+function event.onLogin(player)
+	if not player:getGroup():getAccess() then
```

Comment:
The event should be added to all players, the group check is done in the onThink event, so that if you switch to a online player's group it will still work correctly.